src/appengine/handlers/cron/fuzz_strategy_selection.py

```diff
@@ -35,70 +35,256 @@
 # new_edges metric for each strategy.
 # See https://www.cs.mcgill.ca/~vkules/bandits.pdf for formula.
 
-# TODO(mukundv): Change query to group by strategy selection method once tables
-# are initially populated.
+# TODO(mukundv): Change query once we decide on a temperature parameter and
+# final implementation.
 BANDIT_PROBABILITY_QUERY = """
 SELECT
-  /* Calculate bandit weights from calculated exponential values. */
-  strategy,
-  strategy_exp_medium_temperature / exp_sum_medium_temperature AS bandit_weight_medium_temperature,
-  strategy_exp_high_temperature / exp_sum_high_temperature AS bandit_weight_high_temperature,
-  strategy_exp_low_temperature / exp_sum_low_temperature AS bandit_weight_low_temperature,
-  run_count
-FROM
-  (SELECT
-    EXP(strategy_avg_edges / medium_temperature) AS strategy_exp_medium_temperature,
-    SUM(EXP(strategy_avg_edges / medium_temperature)) OVER() AS exp_sum_medium_temperature,
-    EXP(strategy_avg_edges / high_temperature) AS strategy_exp_high_temperature,
-    SUM(EXP(strategy_avg_edges / high_temperature)) OVER() AS exp_sum_high_temperature,
-    EXP(strategy_avg_edges / low_temperature) AS strategy_exp_low_temperature,
-    SUM(EXP(strategy_avg_edges / low_temperature)) OVER() AS exp_sum_low_temperature,
-    strategy,
-    run_count
-  FROM
-    (SELECT
-      /* Standardize the new edges data and take averages per strategy. */
-      AVG((new_edges - overall_avg_new_edges) / overall_stddev_new_edges) AS strategy_avg_edges,
+  c.strategy,
```

Comment:
The query is very hard to review and will be error prone to edit. At least two sub queries (for medium and high) are exactly the same. I'm not sure about the `low` one, it looks a bit different?

The only pieces that differ in those sub queries are the `medium` / `high` substring and the literal value (`0.5` vs `0.75`).

Can you please move that query into another string with named parameters like `{temperature_type}` and `{temperature_value}`, so that the main logic will be encapsulated at a single place. That string can be appended to the `JOIN` statements with the corresponding values like `.format(temperature_type='medium', temparature_value=0.5)`, etc.